x-pack/plugins/observability_solution/apm_data_access/server/services/get_host_services/index.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { rangeQuery, termQuery } from '@kbn/observability-plugin/server';
+import {
+  AGENT_NAME,
+  HOST_HOSTNAME,
+  HOST_NAME,
+  METRICSET_NAME,
+  SERVICE_NAME,
+} from '@kbn/apm-types/es_fields';
+import { QueryDslQueryContainer } from '@elastic/elasticsearch/lib/api/typesWithBodyKey';
+import { RollupInterval } from '../../../common';
+import { ApmDocumentType } from '../../../common/document_type';
+import type { ApmDataAccessServicesParams } from '../get_services';
+
+const MAX_SIZE = 1000;
+
+export interface HostServicesRequest {
+  filters: Record<string, string>;
+  start: number;
+  end: number;
+  size?: number;
+}
+
+export function createGetHostServices({ apmEventClient }: ApmDataAccessServicesParams) {
+  return async ({ start, end, size = MAX_SIZE, filters }: HostServicesRequest) => {
+    const commonFiltersList: QueryDslQueryContainer[] = [
+      ...rangeQuery(start, end),
+      {
+        exists: {
+          field: SERVICE_NAME,
+        },
+      },
+    ];
+
+    if (filters[HOST_NAME]) {
+      commonFiltersList.push({
+        bool: {
+          should: [
+            ...termQuery(HOST_NAME, filters[HOST_NAME]),
+            ...termQuery(HOST_HOSTNAME, filters[HOST_HOSTNAME]),
+          ],
+          minimum_should_match: 1,
+        },
+      });
+    }
+    // get services from transaction metrics
+    const metricsQuery = await apmEventClient.search('get_apm_host_services_from_metrics', {
+      apm: {
+        sources: [
+          {
+            documentType: ApmDocumentType.TransactionMetric,
+            rollupInterval: RollupInterval.OneMinute,
+          },
+        ],
```

Comment:
Should we use the [`documentSources`](https://github.com/elastic/kibana/blob/main/x-pack/plugins/observability_solution/apm_data_access/server/services/get_host_names/index.ts#L32) data to check which rollupInterval the query needs to use?

Reply:
Yeah, I can try it, I thought it wasn't necessary as the previous query add the 1 minute set for the interval